Projects/19-Captsone_01/ispy1/inferential_statistics.py

```diff
@@ -0,0 +1,132 @@
+    RR_lb: Lower bound of the 95% C.I for the RR
+    RR_ub: Upper bound of the 95% C.I for the RR
+    '''
+    #categorical_predictors = df.columns[2:7]
+    num_pred = len(categorical_predictors)
+    df2 = _pd.DataFrame(np.random.randn(num_pred, 4))
+    df2 = df2.set_index([categorical_predictors])
+    df2.columns = ['p-value', 'Relative_Risk','RR_lb','RR_ub']
+
+    for idx, var in enumerate(categorical_predictors):
+        T = contingency_table(var, outcome, df)
+        _, p , _, _= _chi2( T.values )
+        RR, lb, ub = relative_risk(T.values)
+
+        df2.iloc[idx,0] = p;
+        df2.iloc[idx,1] = RR;
+        df2.iloc[idx,2] = lb;
+        df2.iloc[idx,3] = ub;
+    return df2
+
+# Continous PREDICTORS and CATEGORICAL OUTCOMES
+
+
+def linear_models(df, outcome, predictors, print_results = 1):
+    # create new dataframe with predictors
+    df2 = _pd.DataFrame()
+    df2[predictors] = df[predictors]
+
+    # ad outcome to dataframe with predictors encoded as floating
+    df2[outcome]   = preprocessing.LabelEncoder().fit_transform( df[outcome].values )
+
+    # create formula from strings
+    formula = outcome + "~" + "+".join(predictors )
+
+    # perform ANOVA
+    SPY_lm = ols(formula, data = df2 ).fit()
+    anova = sm.stats.anova_lm(SPY_lm, typ=2) # Type 2 ANOVA DataFrame
+    if print_results == 1:
+        print(15*'---')
+        print(anova)
+        print(15*'---')
+    return anova, SPY_lm
+
+def anova_MRI(outcome, df):
+    mri_predictors= ['MRI_LD_Baseline','MRI_LD_1_3dAC', 'MRI_LD_Int_Reg', 'MRI_LD_PreSurg']
+    results = results = _pd.DataFrame(np.random.random(size=(len(mri_predictors),1)),
+                                          index=mri_predictors, columns=['p-value'])
+    for idx, pred in enumerate(mri_predictors):
+        p = list();     p.append(pred)
+        nova_table, _ = linear_models(df, outcome, p, print_results=0);
+        results.ix[idx] = nova_table['PR(>F)'].values[0]
+
+    f, (ax1, ax2) = plt.subplots(2,2, figsize=(10,10))
+    sns.boxplot(x= outcome, y=mri_predictors[0], data=df, palette="Set3", ax=ax1[0]).set_title('p-value = '+ str(results.values[0]));
+    sns.boxplot(x= outcome, y=mri_predictors[1], data=df, palette="Set3", ax=ax2[0]).set_title('p-value = '+ str(results.values[1]));
+    sns.boxplot(x= outcome, y=mri_predictors[2], data=df, palette="Set3", ax=ax1[1]).set_title('p-value = '+ str(results.values[2]));
+    sns.boxplot(x= outcome, y=mri_predictors[3], data=df, palette="Set3", ax=ax2[1]).set_title('p-value = '+ str(results.values[3]));
+    plt.show()
+    return results
+
+def effect_size(df,predictors, outcome):
+    all_ =  predictors + [outcome]
+    legend = 'Predictor of ' + outcome
+    mean_ = df[all_].groupby( outcome ).mean().values
+    std_ =  np.std( df[predictors].values.flatten() )
+    delta = mean_[0,:] - mean_[1,:]
+    effect_size = _pd.DataFrame(delta/std_)
+
+    effect_size[legend] = predictors
+    effect_size =    effect_size.set_index(legend)
```

Comment:
I found it a little overwhelming in the beginning, but it makes more sense now. I will start using it.
I wrote this project just using atom as text editor.